rpc/send.go

```diff
@@ -151,7 +151,19 @@ func Send(opts Options, method string, addrs []net.Addr, getArgs func(addr net.A
 				log.Infof("%s: sending request to %s: %+v", method, addr, args)
 			}
 			trace.Event(fmt.Sprintf("sending to %s", addr))
-			go sendOneFn(client, opts.Timeout, method, args, getReply(), done)
+
+			reply := getReply()
+
+			// Don't bother firing off a goroutine in the common case where a client
+			// is already healthy.
+			select {
+			case <-client.Healthy():
+				client.Go(method, args, reply, done)
```

Comment:
This bypasses `sendOneFn` which is used to mock out the sending function for one test. Does that test work reliably with this change? Maybe this should only be done when `sendOneFn == sendOne` and that test always uses the slow path. Put comments in `sendOne` that this function is sometimes bypassed so it shouldn't do anything but check the health of the channel and call `client.Go`.